pkg/catalog/catalog.go

```diff
@@ -38,6 +58,34 @@ func GetDevfileIndex(devfileIndexLink string) ([]DevfileIndexEntry, error) {
 	return devfileIndex, nil
 }
 
+var devfileMutex = &sync.Mutex{}
+
+// getDevfileWith retrieves the devfile corresponding to the specified DevfileIndexEntry and creates an associate
+// DevfileComponentType which is then added to the specified DevfileComponentTypeList.
+// This function is meant to be called from go routine, each devfile being processed by its own go routine, using a WaitGroup
+// to wait for all devfiles to be processed before continuing work.
+// If an error occurs, it will be populated in the specified error field.
+func getDevfileWith(devfileIndexEntry DevfileIndexEntry, catalogDevfileList *DevfileComponentTypeList, err error, wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	link := devfileIndexEntry.Links.base + devfileIndexEntry.Links.Link
+	devfile, err := GetDevfile(link)
```

Comment:
same here we need to check the error https://travis-ci.com/github/openshift/odo/jobs/329269812#L271